src/components/translatable/TranslatableTSV.js

```diff
@@ -101,11 +103,45 @@ export default function TranslatableTSV({
     return rowId;
   }, [columnNames, delimiters]);
 
-  const options = {
+  const [options, setOptions] = useState({
     page: 0,
     rowsPerPage: 25,
     rowsPerPageOptions: [10, 25, 50, 100],
-  };
+  });
+
+  const { query } = usePermalinks({});
+
+  useEffect(() => {
+    const exludedFromSearch = ['columns', 'check'];
+
+    if (!query) return;
+
+    //Sets searchText to first searchable param in query string.
+    Object.keys(query).forEach((key) => {
+      if (!exludedFromSearch.includes(key) && !options?.searchText) {
+        setOptions((options) => ({ ...options, searchText: query[key] }));
+      }
+    });
+
+  },[query,options.searchText]);
+
+  const [extraColumns, setExtraColumns] = useState([]);
```

Comment:
Can we move the useState to the top of the component? We try to organize these so that they can be found. We try to keep the useStates that the top.

Reply:
Sure, I try to keep it close to where it is actually used, but I try to fix that later.